.github/workflows/build-and-test.yaml

```diff
@@ -75,13 +81,33 @@ jobs:
         echo "effective_version=${effective_version}" >> "${GITHUB_OUTPUT}"
         echo "repo_version=${src_version}" >> "${GITHUB_OUTPUT}"
         echo "setuptools_version=${setuptools_version}" >> "${GITHUB_OUTPUT}"
+    - name: create commit with effective version
+      run: |
+        git config --global --add safe.directory $PWD
+        git config user.name 'Gardener-CICD Bot'
+        git config user.email 'gardener.ci.user@gmail.com'
+        echo ${{ steps.version.outputs.effective_version }} | .ci/write-version
+        touch /tmp/timestamp-ref
+        git add .
+        if ${{ inputs.release || false }}; then
+          git commit -m "Release ${{ steps.version.outputs.effective_version }}"
+        else
+          git commit -m "Build ${{ steps.version.outputs.effective_version }}"
```

Comment:
I'm not sure if I understood it right: This "build" commit is actually not required, is it? Or more generally, the whole capture/import-commit logic is only required if `inputs.release` is set, correct? If so, I would suggest to drop the condition here and pull it up to the "create commit with effective version" step (same for "capture-commit").

Reply:
@8R0WNI3 : you are correct in that it is not needed in _our_ case (asamof, I also first thought simply not creating this commit in else-case might be the better choice).

However, I figured it might be preferable to consistently create a "commit bearing effective version". Rationale: also for snapshot-builds, there are cases where build-tools need to see changed versionfiles.

OTOH, creating a commit that will not actually be published indeed has the downside of making it difficult (actually impossible w/o looking into buildlogs) to retrospectively map back to build-commit.

So I suppose we also need a second codepath for transporting version-diff (either by leveraging "capture-commit" (but only doing a checkout, rather than a rebase), or by using a different mechanism (e.g. re-create the diff, or expose a patchfile)). Either way, non-release-builds should - different than release-builds - indeed behave differently.

That's unfortunately many different codepaths downstream :-(

Reply:
okay, considering we do not need this transient commit except for release, I agree w/ your proposal of skipping the capturing entirely. I suppose that's a task for another increment.